boilerplate/openshift/golang-osd-operator/standard.mk

```diff
@@ -187,19 +218,46 @@ openapi-generate:
 			-p % \
 			-h /dev/null \
 			-r "-"
+else
+	find ./api -maxdepth 2 -mindepth 1 -type d | xargs -t -n1 -I% \
+		openapi-gen --logtostderr=true \
+			-i % \
+			-o "" \
+			-O zz_generated.openapi \
+			-p % \
+			-h /dev/null \
+			-r "-"
+endif
+	
+
 
 .PHONY: generate
 generate: op-generate go-generate openapi-generate
 
+
 .PHONY: go-build
 go-build: ## Build binary
 	# Force GOOS=linux as we may want to build containers in other *nix-like systems (ie darwin).
 	# This is temporary until a better container build method is developed
 	${GOENV} GOOS=linux go build ${GOBUILDFLAGS} -o ${BINFILE} ${MAINPACKAGE}
 
+# ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
+ENVTEST_K8S_VERSION = 1.23
+
+ENVTEST = $(shell pwd)/bin/setup-envtest
+.PHONY: envtest
+envtest: ## Download envtest-setup locally if necessary.
```

Comment:
@2uasimojo I did some updates:
1. rename envtest to setup-envtest, it's purpose is download/install the setup-envtest binary
2. Using --bin-dir to install the KUBEBUILDER_ASSETS to $(pwd)/bin

> Invoking setup-envtest here in this target rather than inline where it's consumed; setting a variable with the output; and using that variable where it's needed (on L260 in place of the $(shell ...) bit).

Not sure how to expose an variable in one make target, and consumed by another target, so I am not doing this atm.  we only need this when run 'make go-test', so I keeps it there.

Reply:
You can use `eval` to set a variable from within a target. Example:

```makefile
doit: bar
	echo $(FOO)

bar:
	$(eval FOO:=xyz)
```

```
[efried@efried tmp]$ make doit
echo xyz
xyz
```

Reply:
thanks, it works!